### DISCUSSION TITLE
Pimcore Academy: Working with Objects Advanced - How language is determined when calling getters on data objects?

### DISCUSSION THREAD

**Discussion by michaelopre** [pimcore/pimcore#15307] [Q&A / Support] [answered]
Hi, 

I'm struggling again with the following question:

Question 4: How language is determined when calling getters on data objects?
1. By method parameter.
2. By current language of request document.
3. By browser language.
4. By system language of operating system.
5. By random.

When I consider the explanation video, for me the answers 1 and 4 are correct. I also tried 2, because it is plausible that there is a fallback to documents language.

But both selections are wrong!?
In the documentation I don't find a specific page related to that issue.

Can someone explain?

Thank you!
Best,
Michael

**Comment by dvesh3** [comment]
In the 2nd video on localized fields chapter, it is explained how you can get the data by passing the parameter and if the parameter is not passed then fallback language is considered from document language on current request.

**Reply by michaelopre** [reply]
Hmm, maybe I have a misunderstanding in general, but its about data objects and at minute 4:23 he said "Pimcore tries to find the current active system language". That was the reason, why I selected: 
1. By method parameter -> giving the ->getTitle() method a language parameter
2. By current language of the requested document -> because he say that in case of a Controller context, Pimcore tries to get the language of the corresponding document
4. By system language of operating system -> he does it say at minute 4:23 

But it is wrong if I select these 3 answers.

**Reply by michaelopre** [reply]
When I try to track, then in the related get method the getLocalizedFields method tries to get the language by submitted parameter. If no parameter is submitted, the getLocalizedValue method uses the getLanguage method, that returns the language if given.
Otherwise it tries to get the language from the Pimcore::getContainer instance. That means to me, it tries to get the System language or does it tries to get the Document language? Mhh, this is more plausible I think.
If that fails, the exception tries to retun the language from Tool class and the getDefaultLanguage method and this method returns the default language from configuration.

If I consider that, the answer should be 1 and 2 as well, because a "default language" from configuration is not possible to select.

Can someone confirm that?
Best,
Michael

**Reply by michaelopre** [reply]
Ok, I've watched the video again and listened extremely carefully what he said. And he said:

1. by method parameter
2. by system language of the operating system

but it is wrong. Does anybody knows what is the correct answer?

**Comment by michaelopre** [comment, marked as answer]
Ok, even if I disagree, the correct answer is indeed:

1. By parameter
2. By requested document
